packages/json-rpc-engine/src/v2/utils.ts

```diff
@@ -0,0 +1,41 @@
+import {
+  hasProperty,
+  type JsonRpcNotification as BaseJsonRpcNotification,
+  type JsonRpcParams,
+  type JsonRpcRequest as BaseJsonRpcRequest,
+} from '@metamask/utils';
+
+export type JsonRpcNotification<Params extends JsonRpcParams = JsonRpcParams> =
+  BaseJsonRpcNotification<Params>;
+
+export type JsonRpcRequest<Params extends JsonRpcParams = JsonRpcParams> =
+  BaseJsonRpcRequest<Params>;
+
+export type JsonRpcCall<Params extends JsonRpcParams = JsonRpcParams> =
```

Comment:
Nit: "Call" is a bit of a confusing name for a data structure, because a "call" colloquially also refers to a function invocation. 

Could we call this `JsonRpcMessage` instead?

Alternatively we could stick with the spec's definition, and call this `JsonRpcRequest`, and rename `JsonRpcRequest` to.... `JsonRpcRequestWithId`? `NonNotificationJsonRpcRequest`? `JsonRpcRequestWithResponse`? Those are all awful though.